filebeat/input/journald/input.go

```diff
@@ -133,24 +145,28 @@ func (inp *journald) Run(
 	cursor cursor.Cursor,
 	publisher cursor.Publisher,
 ) error {
-	log := ctx.Logger.With("path", src.Name())
-	currentCheckpoint := initCheckpoint(log, cursor)
-
-	reader, err := inp.open(ctx.Logger, ctx.Cancelation, src)
+	logger := ctx.Logger.With("path", src.Name())
+	currentCheckpoint := initCheckpoint(logger, cursor)
+
+	mode := inp.Seek
+	pos := currentCheckpoint.Position
+	reader, err := journalctl.New(
+		logger,
+		ctx.Cancelation,
+		inp.Units,
+		inp.Identifiers,
+		inp.Transports,
+		inp.Matches,
+		mode,
+		pos,
+		inp.Since,
+		src.Name(),
+	)
 	if err != nil {
-		return err
+		return fmt.Errorf("could not start journal reader: %w", err)
```

Comment:
You should set the unit status to failed when running under Elastic Agent.

Reply:
I completely forgot the input can now report status directly to Elastic-Agent.

The current implementation does not support reporting status directly to Elastic-Agent. I believe it is better to implement it as a separate PR and keep this one with the single responsibility of replacing the reader implementation. What do you think?

Reply:
Agree it can be a follow up PR.

Reply:
I created an issue for that already: https://github.com/elastic/beats/issues/40143